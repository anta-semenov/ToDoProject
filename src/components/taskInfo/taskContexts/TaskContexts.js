import React from 'react'
import ImmutablePropTypes from 'react-immutable-proptypes'
import './TaskContexts.less'

export default class TaskContexts extends React.Component {
  render() {
    return (
      <div className='contexts'>
        <div className='contexts__title'>Contexts</div>
        {
          this.props.contexts ?
          <div className='contexts__list'>
            {this.props.contexts.map(context => {
              const active = this.props.taskContexts ? this.props.taskContexts.includes(context.get('id')) : false
              return <div className={`context ${this.props.taskContexts ? active ? 'is-active' : '' : ''}`} onClick={() => this.props.onContextClick(context.get('id'), !active)} >{context.get('title')}</div>
            }
            )}
          </div> :
          <div className={'contexts__empty-list'}>You haven't create any context yet.</div>
        }
      </div>
    )
  }
}

TaskContexts.propTypes = {
  contexts: ImmutablePropTypes.mapOf(
    ImmutablePropTypes.contains({
      id: React.PropTypes.string.isRequired,
      title: React.PropTypes.string.isRequired
    })
  ).isRequired,
  taskContexts: ImmutablePropTypes.iterableOf(React.PropTypes.string),
  onContextClick: React.PropTypes.func.isRequired
}
