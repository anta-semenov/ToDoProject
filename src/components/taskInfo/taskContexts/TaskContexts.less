@import '../../../styles/variables.less';

.contexts {
  position: relative;
  margin: 15px 8px;
}
.contexts__title {
  position: relative;
  margin: 0 0 5px 0;
  font-size: 10px;
  font-weight: 400;
  text-transform: uppercase;
  color: @color-font-light;
}
.contexts__list {
  position: relative;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}
.context {
  position: relative;
  margin: 0 6px 7px 0;
  padding: 2px 8px 1px 8px;
  cursor: pointer;
  transition: @transition-basic;

  border: 1px solid @color-border-gray;
  border-radius: 2px;

  font-size: @font-size-small;
  color: @color-font-light;

  &:hover {
    border-color: @color-border-dark-gray;
    color: @color-font-medium;
  }
  &:active,
  &:focus {
    border-color: @color-accent;
    color: @color-accent;
  }
  &.is-active {
    background-color: @color-accent;
    border-color: @color-accent;
    color: #ffffff;

    &:hover {
      background-color: darken(@color-accent, 10%);
      border-color: darken(@color-accent, 10%);
    }
  }
}
