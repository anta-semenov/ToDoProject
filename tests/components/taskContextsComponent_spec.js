import React from 'react'
import { fromJS, Set } from 'immutable'
import { shallow } from 'enzyme'

import TaskContexts from '../../src/components/taskInfo/taskContexts/TaskContexts'

const contexts = fromJS({
  a21sogy3s0oq: {
    id: 'a21sogy3s0oq',
    title: 'test context 1'
  },
  a23sogy3s0oq: {
    id: 'a23sogy3s0oq',
    title: 'test context 2'
  }
})
const taskContexts = Set(['a23sogy3s0oq'])

describe('TaskContexts component', () => {
  it('Should render component without contexts', () => {
    const contexts = fromJS({})
    const contextsComponent1 = shallow(<TaskContexts onContextClick={() => {}} />)
    const contextsComponent2 = shallow(
      <TaskContexts contexts={contexts} onContextClick={() => {}} />
    )
    expect(contextsComponent1).toMatchSnapshot()
    expect(contextsComponent2).toMatchSnapshot()
  })
  it('Should render component with no task contexts', () => {
    const contextsComponent = shallow(
      <TaskContexts contexts={contexts} onContextClick={() => {}} />
    )
    expect(contextsComponent).toMatchSnapshot()
  })
  it('Should render components with task context', () => {
    const contextsComponent = shallow(
      <TaskContexts contexts={contexts} taskContexts={taskContexts} onContextClick={() => {}} />
    )
    expect(contextsComponent).toMatchSnapshot()
  })
  it('Should handle context click', () => {
    let id
    const onContextClick = newId => (id = newId)
    const contextsComponent = shallow(
      <TaskContexts
        contexts={contexts}
        taskContexts={taskContexts}
        onContextClick={onContextClick}
      />
    )
    expect(id).toBeUndefined()
    contextsComponent
      .find('.context')
      .first()
      .simulate('click')
    expect(id).toBe('a21sogy3s0oq')
  })
})
