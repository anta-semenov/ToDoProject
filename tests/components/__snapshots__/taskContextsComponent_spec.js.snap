// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskContexts component Should render component with no task contexts 1`] = `
<div
  className="contexts"
>
  <div
    className="contexts__title"
  >
    Contexts
  </div>
  <div
    className="contexts__list"
  >
    <div
      className="context "
      key="a21sogy3s0oq"
      onClick={[Function]}
    >
      test context 1
    </div>
    <div
      className="context "
      key="a23sogy3s0oq"
      onClick={[Function]}
    >
      test context 2
    </div>
  </div>
</div>
`;

exports[`TaskContexts component Should render component without contexts 1`] = `
<div
  className="contexts"
>
  <div
    className="contexts__title"
  >
    Contexts
  </div>
  <div
    className="contexts__empty-list"
  >
    You haven't create any context yet.
  </div>
</div>
`;

exports[`TaskContexts component Should render component without contexts 2`] = `
<div
  className="contexts"
>
  <div
    className="contexts__title"
  >
    Contexts
  </div>
  <div
    className="contexts__empty-list"
  >
    You haven't create any context yet.
  </div>
</div>
`;

exports[`TaskContexts component Should render components with task context 1`] = `
<div
  className="contexts"
>
  <div
    className="contexts__title"
  >
    Contexts
  </div>
  <div
    className="contexts__list"
  >
    <div
      className="context "
      key="a21sogy3s0oq"
      onClick={[Function]}
    >
      test context 1
    </div>
    <div
      className="context is-active"
      key="a23sogy3s0oq"
      onClick={[Function]}
    >
      test context 2
    </div>
  </div>
</div>
`;
